### PR TITLE
feat: add BitBake

### PR DESCRIPTION
**Pull request by amaanq**
Just a note that this language allows for python function definitions annoyingly, hence the major overlap

### CODE REVIEW

File: queries/bitbake/indents.scm [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+ (#set! indent.immediate 1))
+((try_statement) @indent.begin
+ (#set! indent.immediate 1))
+(ERROR "try" ":" @indent.begin (#set! indent.immediate 1))
+((python_function_definition) @indent.begin
+ (#set! indent.immediate 1))
+((function_definition) @indent.begin
+ (#set! indent.immediate 1))
+((anonymous_python_function) @indent.begin
+ (#set! indent.immediate 1))
```
Comment (lucario387) [inline queries/bitbake/indents.scm:46]: From bitbake docs (is it the doc? https://docs.yoctoproject.org/bitbake/bitbake-user-manual/bitbake-user-manual-metadata.html#functions)

Then functions like `func(){}` are allowed, which means normal C-like functions. If this is the normal function, then it probably doesn't need `indent.immediate`, while python style's function should have it

Comment (amaanq) [reply queries/bitbake/indents.scm:46]: yeah you're right :sweat_smile: 

only python_function should retain that

**Comment by lucario387** [comment]
According to the doc, this is a valid if-else sequence:
```
if [condition] ; then
  # something
else 
  # something else
fi
```
This looks more bash-like than python-like

**Comment by amaanq** [comment]
> According to the doc, this is a valid if-else sequence:
> 
> ```
> if [condition] ; then
>   # something
> else 
>   # something else
> fi
> ```
> 
> This looks more bash-like than python-like

https://github.com/amaanq/tree-sitter-bitbake/blob/43e55585c1fb60b9da3b7da821f584cf06d6d107/grammar.js#L219

It is, and the node is marked as shell_content and is injected w/ bash

**Comment by amaanq** [comment]
I'm also thinking about getting rid of the python embedding and just considering every line that doesn't start at col 0 as python/bash code, but the spec isn't clear about how that works :smile:

**Comment by ObserverOfTime** [comment]
Oh boy, another Python-like build tool.

File: queries/bitbake/folds.scm [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  (list_comprehension)
+  (set_comprehension)
+  (dictionary_comprehension)
+
+  (tuple)
+  (list)
+  (set)
+  (dictionary)
+
+  (string)
```
Comment (ObserverOfTime) [inline queries/bitbake/folds.scm:27]: Can non-Python strings be multiline?

Comment (amaanq) [reply queries/bitbake/folds.scm:27]: yes

**Pull request merged**